### ISSUE TITLE
Keep focus if overlay is closed by clicking outside it

### ISSUE THREAD

**Issue by vicsstar** [vaadin/vaadin-date-picker#475]
### Browsers Affected

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE 11
- [ ] iOS Safari
- [ ] Android Chrome

**Comment by jouni** [comment]
This is basically the same issue as https://github.com/vaadin/vaadin-dialog/issues/98, and apparent we could fix this by using the `overlay.restoreFocusOnClose` property.

**Comment by jouni** [comment]
It’s not exactly the same as the dialog issue, as focus is restored to the input if you close the overlay using the Esc key. But when the overlay is modal, clicking outside it is considered as the same action.

**Comment by jouni** [comment]
For some reason, Select behaves different. Focus is restored to the field when the overlay is closed. Combo box works the same as date picker.

@rolfsmeds, should we prioritize this issue, as it seems a little inconsistent?

**Comment by rolfsmeds** [comment]
Question is: which is correct? Should clicks outside **non-modal** overlays return focus to the field? What if I click something interactive, like a Button? I'd expect focus to stay on the button in that case.

**Comment by jouni** [comment]
Non-modals are obvious, in my opinion: whatever you click gets the focus. If the overlay has “close on outside click” it’ll close, otherwise it’ll stay open.

Note, that all of our overlays are modal, even though they do not always show the curtain/backdrop.

**Comment by jouni** [comment]
But your question is still valid, I think.

On macOS, the native `<select>` regains the focus when you click outside the overlay to dismiss it.
